Source/Moya.swift

```diff
@@ -65,12 +87,16 @@ public class MoyaProvider<Target: TargetType> {
     }
     
     /// Returns an Endpoint based on the token, method, and parameters by invoking the endpointsClosure.
-    public func endpoint(token: Target) -> Endpoint<Target> {
+    public func endpoint(token: TargetType) -> Endpoint<TargetType> {
         return endpointClosure(token)
     }
     
     /// Designated request-making method. Returns a Cancellable token to cancel the request later.
     public func request(target: Target, completion: Moya.Completion) -> Cancellable {
+        guard let target = target.resource else {
```

Comment:
Of course! So the general idea of `ServiceType` is that this is the core of **StructsAPI**. Every enum that wants to use Moya with structs would have to conform to `ServiceType`. We then believe that when you conform to that protocol, you know what are you doing (I mean that is the risk I was talking about on hangout). So as a follow-up, I continue to believe the user and I try to fetch the associated object as it is a Resource (TargetType). But! I've made `TargetType` to conform to `ServiceType` as a default. So if you are using **EnumAPI**, it basically means that we don't need to look for associated object, we are the associated object, we are a TargetType (Resource). 

I don't know if that seems good to you, but let me know, I will try harder express my mind 🤕